lib/cli.rb

```diff
@@ -77,15 +79,30 @@ def define_options(parser)
                   'Internal use only',
                   &method(:workspace_path=))
       end
-      # rubocop:enable Metrics/MethodLength
     end
 
     # class TestOptions
-    class TestOptions
-      attr_accessor :platform, :drivers, :driver_path, :commit, :svvp, :dump,
-                    :gthb_context_prefix, :gthb_context_suffix, :playlist, :select_test_names,
-                    :reject_test_names, :reject_report_sections, :boot_device,
-                    :allow_test_duplication, :manual, :package_with_playlist
+    class TestOptions < T::Struct
+      include T::Sig
+
+      prop :platform, T.nilable(String)
+      prop :drivers, T.nilable(T::Array[String])
+      prop :driver_path, T.nilable(String)
+      prop :commit, T.nilable(String)
+      prop :svvp, T.nilable(T::Boolean)
+      prop :dump, T.nilable(T::Boolean)
+      prop :gthb_context_prefix, T.nilable(String)
+      prop :gthb_context_suffix, T.nilable(String)
+      prop :playlist, T.nilable(String)
+      prop :select_test_names, T.nilable(String)
+      prop :reject_test_names, T.nilable(String)
+      prop :reject_report_sections, T::Array[String], default: []
+      prop :boot_device, T.nilable(String)
+      prop :allow_test_duplication, T.nilable(T::Boolean)
+      prop :manual, T.nilable(T::Boolean)
+      prop :package_with_playlist, T.nilable(T::Boolean)
```

Comment:
Don't see any reason to make boolean fields nilable. String or Array can be nilable because we have logic to check is field specified or not. But in the case of a boolean, we always check true or false.